### ISSUE TITLE
Using Resources.LoadAll

### ISSUE THREAD

**Issue by Navil** [SolidAlloy/GenericUnityObjects#40]
Hey,

pretty much new to this package (and C#, but not programming itself). I have that usecase where I need my SO to be generic, but Unity does not really allow that.
As I tried setting up an event driven Architecture, where each Ability is tied to a specific CombatEvent (e.g. StartOfRound, OnAttack, ...), I tried the following signature:
`public abstract class AbilityData<T> : GenericScriptableObject where T: CombatEvent`

However, calling
`Resources.LoadAll<AbilityData<CombatEvent>>("Abilities")` does not yield me any results. Is there any replacement function to be used, so.I can load all my GenericScriptableObjects from a folder?

**Comment by CraigGraff** [comment]
@Navil Do you have your GenericScriptableObject assets in a Resources folder (or subfolder of that)?

**Comment by Navil** [comment]
Yes, I do. It is working if I use normal ScriptableObjects and remove the generic elements.
